### PR TITLE
[#OCD-3261] feat!: Update removeCriteriaJob for ERD Phase 2 requirements

### PR DESCRIPTION
**Pull request by andlar**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/criteriaremoval/RemoveCriteriaJob.java [hunk 1]
```diff
@@ -30,18 +48,23 @@ public RemoveCriteriaJob() throws Exception {
...
+                    String[] criterionElements = criteria.split(":");
+                    CertificationCriterionDTO certDto = certCriteriaDao.getByNumberAndTitle(criterionElements[0], criterionElements[1]);
+                    certDto.setRemoved(true);
+                    certCriteriaDao.update(certDto);
+                    LOGGER.info("Updated criteria " + criteria);
+                } catch (final Exception ex) {
+                    LOGGER.error("Exception updating criteria " + criteria, ex);
+                }
             }
+            CacheManager.getInstance().clearAll();
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/criteriaremoval/RemoveCriteriaJob.java:76]: Can you also clear the shared store here? I suspect this job will be run after the shared store is deployed.

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/criteriaremoval/RemoveCriteriaJob.java (hunk 1 above)
Comment (andlar) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/criteriaremoval/RemoveCriteriaJob.java:76]: Can I? Or does that depend on OCD-3949 going to STG? Also not sure if that branch even _has_ a "clear all store" method

Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/criteriaremoval/RemoveCriteriaJob.java:76]: Don't know! Question for @tmy1313 . I believe it needs it at some point anyway so I want to leave this comment here but approve the feature going forward if the clearing code is not available yet. If it depends on OCD-3949 then you'll need to make a commit with that clearing code when this goes to QA I guess.

Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/criteriaremoval/RemoveCriteriaJob.java:76]: My thought would be to add a checkbox to OCD-3949 that indicates that we need to clear the Shared Listing Store when this job is run and then implement the change when this ticket goes to STG.  I would leave the code for this ticket alone for now.  But this does leave implications for testing this change - how do we test this change in any environment where OCD-3949 has already been implemented?  We probably need to discuss...

**Review by kekey1 — APPROVED** [review]

**Review by tmy1313 — APPROVED** [review]

**Pull request merged**